api/src/main/java/com/github/eyefloaters/console/api/model/PartitionInfo.java

```diff
@@ -35,39 +36,46 @@ public PartitionInfo(int partition, Node leader, List<Node> replicas, List<Node>
     }
 
     public static PartitionInfo fromKafkaModel(org.apache.kafka.common.TopicPartitionInfo info) {
-        Node leader = Node.fromKafkaModel(info.leader());
         List<Node> replicas = info.replicas().stream().map(Node::fromKafkaModel).toList();
-        List<Node> isr = info.isr().stream().map(Node::fromKafkaModel).toList();
-        return new PartitionInfo(info.partition(), leader, replicas, isr);
+        Integer leader = info.leader().id();
+        List<Integer> isr = info.isr().stream().map(org.apache.kafka.common.Node::id).toList();
+        return new PartitionInfo(info.partition(), replicas, leader, isr);
     }
 
-    static Either<OffsetInfo, Error> offsetOrError(Either<OffsetInfo, Throwable> offset) {
+    static <P> Either<P, Error> primaryOrError(Either<P, Throwable> offset, String message) {
```

Comment:
The name of the `offset` parameter should be `primary`?

Reply:
Fixed this - renamed `offset` to just a generic name `either`, since `primary` would represent only one side of the `Either`.